test/saithrift/tests/sail3.py

```diff
@@ -2025,3 +2025,632 @@ def runTest(self):
+                if (i % 100) == 0:
+                    print "Sending %s packet " % (i)
+                pkt = simple_tcp_packet(
+                                eth_dst=router_mac,
+                                eth_src=src_mac,
+                                ip_dst=ip_dst,
+                                ip_src=ip_src,
+                                tcp_sport=sport,
+                                tcp_dport=dport,
+                                ip_id=106,
+                                ip_ttl=64)
+                exp_pkt1 = simple_tcp_packet(
+                                eth_dst='00:11:22:33:44:55',
+                                eth_src=router_mac,
+                                ip_dst=ip_dst,
+                                ip_src=ip_src,
+                                tcp_sport=sport,
+                                tcp_dport=dport,
+                                ip_id=106,
+                                ip_ttl=63)
+                exp_pkt2 = simple_tcp_packet(
+                                eth_dst='00:11:22:33:44:56',
+                                eth_src=router_mac,
+                                ip_dst=ip_dst,
+                                ip_src=ip_src,
+                                tcp_sport=sport,
+                                tcp_dport=dport,
+                                ip_id=106,
+                                ip_ttl=63)
+
+                exp_pkt3 = simple_tcp_packet(
+                                eth_dst='00:11:22:33:44:57',
+                                eth_src=router_mac,
+                                ip_dst=ip_dst,
+                                ip_src=ip_src,
+                                tcp_sport=sport,
+                                tcp_dport=dport,
+                                ip_id=106,
+                                ip_ttl=63)
+
+                send_packet(self, 2, str(pkt))
+                rcv_idx = verify_any_packet_any_port(self, [exp_pkt1, exp_pkt2, exp_pkt3], [0, 1, 3])
+                count[rcv_idx] += 1
+                sport += 1
+                dport += 1
+
+            for i in range(0, 3):
+                print "Packet received on port %s  >> %s" % (i, count[i])
+                self.assertTrue((count[i] >= ((max_itrs / 3) * 0.8)),
+                                "Not all paths are equally balanced, %s" % count)
+            print "All paths are  balanced with three members"
+            # L3IPv4EcmpGroupMemberTest_II test case
+            # Remove the nhop bound with router interface 1, and verify the traffic flows towards the ECMP route.
+            self.client.sai_thrift_remove_next_hop_group_member(nhop_gmember1)
+
+            count = [0, 0]
+            max_itrs = 101
+            itr_count = 0
+
+            print "sending %s packets to verify distribution on 2 ports after removing one ecmp group member" %(max_itrs - 1 )
+
+            for i in range(1, max_itrs):
```

Comment:
The for loop is repeated and looks like it is common code. Can you check if it can be put into one function to avoid duplication?

Reply:
Addressed comments